staging/src/k8s.io/apiserver/pkg/server/openapi/openapi.go

```diff
@@ -64,14 +92,36 @@ func RegisterOpenAPIService(servePath string, webServices []*restful.WebService,
 		return err
 	}
 
-	mux.UnlistedHandleFunc(servePath, func(w http.ResponseWriter, r *http.Request) {
-		resp := restful.NewResponse(w)
-		if r.URL.Path != servePath {
-			resp.WriteErrorString(http.StatusNotFound, "Path not found!")
-		}
-		// TODO: we can cache json string and return it here.
-		resp.WriteAsJson(o.swagger)
-	})
+	type fileInfo struct {
+		ext  string
+		data []byte
+		mime string
+	}
+
+	files := []fileInfo{
+		{".json", o.swaggerBytes, MIME_JSON},
+		{".pb", o.swaggerPb, MIME_PB},
+		{".pb.gz", o.swaggerPbGz, MIME_PB_GZ},
+		{".json.sha512", o.swaggerHash[:], MIME_BINARY},
+		{".pb.sha512", o.swaggerPbHash[:], MIME_BINARY},
+		{".pb.gz.sha512", o.swaggerPbGzHash[:], MIME_BINARY},
+	}
+
+	for _, file := range files {
+		path := servePathBase + file.ext
+		mime := file.mime
+		data := file.data
+		mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+			resp := restful.NewResponse(w)
+			if r.URL.Path != path {
+				resp.WriteErrorString(http.StatusNotFound, "Path not found!")
```

Comment:
I think it will and I don't think this check serve anything here, just looked at other HandleFuncs in our code and they all checked this, I will look into this more and remove this if statement if it doesn't help in any way.